File: services/shared/domain/chunk.go
package domain

type Chunk struct {
	ChunkID                string `json:"chunk_id"`
	RepoURL                string `json:"repo_url"`
	FileName               string `json:"file_name"`
	Branch                 string `json:"branch"`
	CommitSHA              string `json:"commit_sha"`
	SnapshotID             string `json:"snapshot_id"`
	FilePath               string `json:"file_path"`
	DocumentType           string `json:"document_type"`
	Language               string `json:"language"`
	Content                string `json:"content"`
	ContentHash            string `json:"content_hash"`
	SymbolName             string `json:"symbol_name"`
	SymbolType             string `json:"symbol_type"`
	StartLine              int    `json:"start_line"`
	EndLine                int    `json:"end_line"`
	StartChar              int    `json:"start_char"`
	EndChar                int    `json:"end_char"`
	PrevChunkID            string `json:"prev_chunk_id,omitempty"`
	NextChunkID            string `json:"next_chunk_id,omitempty"`
	Status                 string `json:"status"`
	TextSearchEngineStatus string `json:"textsearchengine_status"`
	VectorDbStatus         string `json:"vectordb_status"`
	GraphDbStatus          string `json:"graphdb_status"`
	CreatedAt              int64  `json:"created_at"`
}

File: services/query-service/app/models/chunk.py
from __future__ import annotations

from typing import Any

from app.models.retrieval import RetrievalHit


def retrieval_hit_from_chunk(
    chunk: dict[str, Any],
    *,
    score: float = 0,
    source: str,
) -> RetrievalHit | None:
    chunk_id = str(chunk.get("chunk_id") or "").strip()
    if not chunk_id:
        return None

    return RetrievalHit(
        chunk_id=chunk_id,
        repo_url=optional_string(chunk.get("repo_url")),
        branch=optional_string(chunk.get("branch")),
        snapshot_id=optional_string(chunk.get("snapshot_id")),
        file_path=optional_string(chunk.get("file_path")),
        language=optional_string(chunk.get("language")),
        symbol_name=optional_string(chunk.get("symbol_name")),
        symbol_type=optional_string(chunk.get("symbol_type")),
        start_line=optional_int(chunk.get("start_line")),
        end_line=optional_int(chunk.get("end_line")),
        content=optional_string(chunk.get("content")),
        prev_chunk_id=optional_string(chunk.get("prev_chunk_id")),
        next_chunk_id=optional_string(chunk.get("next_chunk_id")),
        score=score,
        sources=[source],
    )


def optional_string(value: Any) -> str | None:
    if value is None:
        return None
    text = str(value)
    return text if text else None


def optional_int(value: Any) -> int | None:
    if value is None or value == "":
        return None
    return int(value)
